parcel/parcel.jl

```diff
@@ -0,0 +1,216 @@
+import OrdinaryDiffEq as ODE
+import CairoMakie as MK
+
+import Thermodynamics as TD
+import CloudMicrophysics as CM
+import CLIMAParameters as CP
+
+const CMT = CM.CommonTypes
+const CMO = CM.Common
+const CMI = CM.HetIceNucleation
+const CMP = CM.Parameters
+
+FT = Float64
+
+# Boiler plate code needed to have access to model parameters and constants
+include(joinpath(pkgdir(CM), "test", "create_parameters.jl"))
+toml_dict = CP.create_toml_dict(FT; dict_type = "alias")
+const param_set = cloud_microphysics_parameters(toml_dict)
+const thermo_params = CMP.thermodynamics_params(param_set)
+
+"""
+    ODE problem definition
+"""
+function cirrus_box(dY, Y, p, t)
+
+    # Our state vector
+    S_i = Y[1]        # supersaturation over ice
+    N_act = Y[2]      # number concentration of activated particles
+    p_a = Y[3]        # pressure
+    T = Y[4]          # temperature
+    q_vap = Y[5]      # vapor specific humidity
+    q_ice = Y[6]      # ice specific humidity
+    N_aerosol = Y[7]  # nuber concentration of interstitial aerosol
+
+    # Constants
+    R_v = CMP.R_v(param_set)
+    grav = CMP.grav(param_set)
+    ρ_ice = CMP.ρ_cloud_liq(param_set)
+
+    # Get phase partition and create thermo state.
+    q = TD.PhasePartition(q_vap + q_ice, FT(0), q_ice)
+    ts = TD.PhaseNonEquil_pTq(thermo_params, p_a, T, q)
+
+    # Constants and variables that depend on the moisture content
+    R_a = TD.gas_constant_air(thermo_params, q)
+    cp_a = TD.cp_m(thermo_params, q)
+    L_subl = TD.latent_heat_sublim(thermo_params, T)
+    ρ = TD.air_density(thermo_params, ts)
+
+    # We are solving for both q_ and supersaturation. This is unneccessary.
+    # I'm checking here if we are consistent. We should decide
+    # if we want to solve for S (as in the cirrucs box model) or for q_
+    # which is closer to what ClimaAtmos is doing.
+    e_si = TD.saturation_vapor_pressure(thermo_params, T, TD.Ice())
+    e = S_i * e_si
+    e_check = q_vap * p_a * R_v / R_a
+    @assert isapprox(e, e_check; rtol = 0.01)
+
+    # Adiabatic parcel coefficients
+    a1 = L_subl * grav / cp_a / T^2 / R_v - grav / R_a / T
+    a2 = p_a / e_si * R_v / R_a
+    a3 = L_subl^2 / R_v / T^2 / cp_a
+
+    # Activating new crystals
+    AF = CMI.dust_activated_number_fraction(S_i, T, CMT.DesertDustType())
+    τ_relax = const_dt
```

Comment:
Maybe pass `const_dt` to the function as a parameter through `p` instead of using it as a global variable.